Add table helper functions (#31)
infinum/ngx-nuts-and-bolts: libs/ngx-nuts-and-bolts/src/lib/helpers/table-state/README.md

# Table state helpers

> Being able to take out the pagination, sorting and filtering information from the memory of the application alleviates some of the problems (link sharing, refreshing pages) which are present when saving that data in-memory. For that purpose this library exposes functions which cover common use cases regarding table state manipulation.

Reviewer: ```suggestion
Being able to take out the pagination, sorting and filtering information from the memory of the application alleviates some of the problems (link sharing, refreshing pages) which are present when saving that data in-memory. For this purpose, this library exposes functions which cover common use cases regarding table state manipulation.
```

## 1. Features

- Creation of pagination/sorting/filtering observables which enable consumers to react to pagination/sorting/filtering query paramater change events.
- Saving table state information (pagination/sorting/filtering) into the query parameters.

## 2. Usage

> Helper functions can be used individually or as example below shows with other helper functions and a little bit of RxJS magic which then reacts and re-fethces data based on changes in any one of the passed observables.

Reviewer: ```suggestion
Helper functions can be used individually or, as the example below shows, with other helper functions and a little bit of RxJS magic which reacts to state changes and re-fetches the data,
```

```ts

interface ITemplateData{
	data: Data;
	pagination: IPageInfo;
	sort: ISortInfo;
	filters: TFilterValue;
}

export class MyComponent extends LoadingStateComponent {
	public readonly templateData$ = this.createTemplateDataObservable();

	constructor(
		private readonly dataService: DataService,
		private readonly route: ActivatedRoute,
		private readonly router: Router
	) {}

	private createTemplateDataObservable(): Observable<ITemplateData> {
		const pagination$ = createPaginationObservable();
		const sort$ = createSortingObservable();
		const filters$ = createFilteringObservable();

> 		return combineLatest([pagination$, sort$, filters$, this.loadingTrigger$]).pipe(

Reviewer: Since we removed distinct and debounce from the lib, this example should be updated to something like this:

```ts
import { isEqual } form 'lodash'; // or some other way of comparing changes.

const tableState$ = combineLatest([pagination$, sort$, filters$]).pipe(
  debounceTime(500), // optional, might depend on the use case whether you want this or not
  distinctUntilChanged(isEqual) // recommended
)

return combineLatest([tableState$, this.loadingTrigger$]).pipe(...)
